transcoder/ffmpeg_segment_failcase_test.go

```diff
@@ -0,0 +1,266 @@
+		} else {
+			fmt.Println("Failed in parsing .txt file:", i+1, "-", file)
+		}
+		txtFile.Close()
+	}
+	return nil
+}
+
+func checkTranscodingFailCase(t *testing.T, inputs [][]string, accel ffmpeg.Acceleration, outdir string, outcsv string) (int, error) {
+
+	failcount := len(inputs)
+	fwriter, err := os.Create(outcsv)
+	defer fwriter.Close()
+	if err != nil {
+		return failcount, err
+	}
+	csvrecorder := csv.NewWriter(fwriter)
+	defer csvrecorder.Flush()
+	//write header
+	columnheader := []string{"video-filepath", "transcoding-pofile", "source-info", "source-path", "error-string"}
+	_ = csvrecorder.Write(columnheader)
+
+	ffmpeg.InitFFmpegWithLogLevel(ffmpeg.FFLogWarning)
+
+	for i, indata := range inputs {
+
+		profiles := parseProfiles(indata[1])
+		tc := ffmpeg.NewTranscoder()
+
+		profs2opts := func(profs []ffmpeg.VideoProfile) []ffmpeg.TranscodeOptions {
+			opts := []ffmpeg.TranscodeOptions{}
+			for n, p := range profs {
+				oname := fmt.Sprintf("%s/%s_%d_%d.ts", outdir, p.Name, n, i)
+				muxer := "mpegts"
+
+				o := ffmpeg.TranscodeOptions{
+					Oname:        oname,
+					Profile:      p,
+					Accel:        accel,
+					AudioEncoder: ffmpeg.ComponentOptions{Name: "copy"},
+					Muxer:        ffmpeg.ComponentOptions{Name: muxer},
+				}
+				opts = append(opts, o)
+			}
+			return opts
+		}
+		in := &ffmpeg.TranscodeOptionsIn{
+			Fname: indata[0],
+			Accel: accel,
+		}
+		out := profs2opts(profiles)
+		_, err = tc.Transcode(in, out)
+
+		wrecord := indata
+		if err == nil {
+			fmt.Println("Succeed in transcoding:", i, profiles, indata[0])
+			wrecord = append(wrecord, "success")
+			failcount--
+		} else {
+			fmt.Println("Failed in transcoding:", i, profiles, indata[0])
+			wrecord = append(wrecord, err.Error())
+		}
+		csvrecorder.Write(wrecord)
+		tc.StopTranscoder()
+	}
+
+	return failcount, nil
+}
+
+func TestNvidia_CheckFailCase(t *testing.T) {
+	indir := os.Getenv("FAILCASE_PATH")
+	if indir == "" {
+		t.Skip("Skipping FailCase test; no FAILCASE_PATH set for checking fail case")
+	}
+
+	outcsv := "result.csv"
+	accel := ffmpeg.Nvidia
+	outdir, err := ioutil.TempDir("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(outdir)
+
+	var infiles []string
+	err = filepath.Walk(indir, selectFile(&infiles))
+	if len(infiles) == 0 || err != nil {
+		t.Skip("Skipping FailCase test. Can not collect fail case files")
+	}
+	//srarting parse .txt files and write meta csv file for test
+	fmt.Println("Start parsing .txt based files.")
+	var inTparams [][]string
+	err = parsingAndStore(t, infiles, outdir, &inTparams)
+
+	if err != nil {
+		t.Error("Failed in parsing .txt based files.")
+	}
+	fmt.Println("Start transcoding from failed source files.")
+
+	failcount, lasterr := checkTranscodingFailCase(t, inTparams, accel, outdir, outcsv)
+
+	if lasterr != nil {
+		t.Error("Failed in checking fail case files.")
+	}
+	fmt.Println("Succeed in fail check- real failed:", failcount, " / total:", len(inTparams))
```

Comment:
Could we have a more concise message, like: "Test completed, really failed: XXX / total: XXX"